packages/host/app/resources/module-contents.ts

```diff
@@ -97,52 +115,50 @@ export class ModuleContentsResource extends Resource<Args> {
       moduleSyntax,
       exportedCardsOrFields,
     );
-    this._declarations = moduleSyntax.declarations.reduce(
-      (acc: ModuleDeclaration[], value: Declaration) => {
-        if (isPossibleCardOrFieldClass(value)) {
-          const cardOrField = exportedCardsOrFields.find(
-            (c) => c.name === value.localName,
-          );
-          let loader = Loader.getLoaderFor(cardOrField);
-          if (cardOrField && loader !== undefined) {
-            return [
-              ...acc,
-              {
-                ...value,
-                cardOrField,
-                cardType: getCardType(
-                  this,
-                  () => cardOrField as typeof BaseDef,
-                  () => loader as Loader,
-                ),
-              } as CardOrField & Partial<PossibleCardOrFieldClass>,
-            ];
-          } else {
-            if (localCardsOrFields.has(value)) {
-              let cardOrField = localCardsOrFields.get(value) as typeof BaseDef;
-              // we don't check for loader here because cards or fields not defined in module will not have a loader
-              if (cardOrField) {
-                return [
-                  ...acc,
-                  {
-                    ...value,
-                    cardOrField,
-                    cardType: getCardType(this, () => cardOrField),
-                  } as CardOrField & Partial<PossibleCardOrFieldClass>,
-                ];
-              }
-            }
+    this._declarations = [];
+    moduleSyntax.declarations.forEach((value: Declaration) => {
+      if (value.type === 'possibleCardOrField') {
+        let cardOrField = value.exportedAs
+          ? exportedCardsOrFields.get(value.exportedAs)
+          : localCardsOrFields.get(value);
+        if (cardOrField !== undefined) {
+          this._declarations.push({
+            ...value,
+            cardOrField,
+            cardType: getCardType(this, () => cardOrField as typeof BaseDef),
+          } as CardOrFieldDeclaration);
+          return;
+        }
+        // case where things statically look like cards or fields but are not
+        if (value.exportedAs !== undefined) {
+          this._declarations.push({
+            localName: value.localName,
+            exportedAs: value.exportedAs,
+            path: value.path,
+            type: 'class',
+          } as ClassDeclaration);
+        }
+      } else if (value.type === 'reexport') {
+        let cardOrField: typeof BaseDef | undefined;
+        if (value.exportedAs) {
+          let foundCardOrField = exportedCardsOrFields.get(value.exportedAs);
+          if (foundCardOrField) {
+            cardOrField = foundCardOrField;
+          }
+          if (cardOrField !== undefined) {
+            this._declarations.push({
+              ...value,
+              cardOrField,
+              cardType: getCardType(this, () => cardOrField as typeof BaseDef),
+            } as CardOrFieldReexport);
```

Comment:
new case for re-export. Since we capture card or field of exports, we have to track it in module contents